Add common build tool (#633)
GDATASoftwareAG/vaas: Readme.md

The following table shows the functionality supported by each SDK:

|Functionality|Rust|Java|PHP|TypeScript|.NET|Python|Ruby|Golang|C++|
|---|---|---|---|---|---|---|---|---|---|
|Check SHA256|&#9989;|&#9989;|&#9989;|&#9989;|&#9989;|&#9989;|&#9989;|&#9989;|&#9989;|
|Check SHA256 list|&#9989;|&#9989;|&#10060;|&#9989;|&#9989;|&#10060;|&#10060;|&#9989;|&#10060;|
|Check URL|&#9989;|&#9989;|&#9989;|&#9989;|&#9989;|&#9989;|&#9989;|&#9989;|&#10060;|
|Check file|&#9989;|&#9989;|&#9989;|&#9989;|&#9989;|&#9989;|&#9989;|&#9989;|&#9989;|
|Check file list|&#9989;|&#9989;|&#10060;|&#9989;|&#9989;|&#10060;|&#10060;|&#9989;|&#10060;|
|Custom Guids for tracability on user side|&#10060;|&#10060;|&#9989;|&#10060;|&#10060;|&#9989;|&#10060;|&#10060;|&#10060;|

## Integration Ideas for Malware Detection trough VaaS
You can use VaaS to create various applications that scan for malicious content with a few lines of code. Here are some examples:

- [WordPress Plugin](https://wordpress.org/plugins/gdata-antivirus/) to scan for malware in uploaded files

> - [Nextcloud App](https://apps.nextcloud.com/apps/gdatavaas) to scan files in your Nextcloud instance

Reviewer: thank you for adding those!

## Build & Test

The easiest way to build and test the SDKs is with the [Nix Package Manager](https://nixos.org/download/#download-nix) and the provided [Just](https://github.com/casey/just) file. Nix will take care of all dependencies and Just provides a simple interface to run the most common tasks.

To build and test the SDKs, run the following command:

```bash
# switch into a development shell with all dependencies installed.
# This will not alter your system, but provide a shell with all necessary tools.
nix develop

# Now use the Just tool to run the most common tasks
just -l # list all available tasks

# Just Examples
# Run the tests for the Rust SDK
just test-rust
```

There are `test-*`, `build-*`, `clean-*` and `release-*` tasks for each SDK, with the exception of Python and PHP, where no build task is available. You can also run the tests for all SDKs with `just test-all`. A `build-all` and `clean-all` task is available as well.

> The `release-*` task triggers a Github Action to build and release a new version of the specified SDK. It needs a version number as an argument, which is used to tag the release. The version number should follow the [Semantic Versioning](https://semver.org/) scheme.

Reviewer: does this actually create a github release?

Author: It's just a shortcut for `git tag -a v1.2.3 -m "Release ..." && git push origin v1.2.3`. That's how I released new versions of the SDKs. If a Github release is created is up to the CI pipeline, which I did not touch. So I'm not sure if the pipeline creates a release for the artifacts on push.

Reviewer: Ah I see. We actually do it the other way around. Meaning we create a github-release in the UI and with the release we create the tag. We could discuss doing it your way and implement a github-release step into the pipelines.

Author: Makes sense. But for this PR I would not change anything as so far, both approaches seem to work fine.

```bash
# Example: Release the Rust SDK with version 0.1.0
just release-rust 0.1.0
```

> As the SDKs need credentials to authenticate against the VaaS API, you need to provide them in a `.env` file. Copy your `.env` file in the root of the project. The C++ SDK needs special credentials, which need to be provided in the `.cpp.env` file.

Reviewer: ```suggestion
As the SDKs need credentials to authenticate to the VaaS API. You need to provide them in a `.env` file. Copy your `.env` file into the root directory of the project. The C++ SDK needs special credentials, which you can provide in a `.cpp.env` file.
```

Reviewer: Maybe an Example of the .env-Files would be great.
